src/mprismetadata.cpp

```diff
@@ -19,6 +17,210 @@
+
+    Maps top MPRIS property 'mpris.length'.
+*/
+
+/*!
+    \qmlproperty var MprisMetaData::coverArtUrlSmall
+    \brief Small version of a cover image for the media
+
+    An url for a cover image of the media.
+    This field is only provided for QtMultimedia compatibility, the
+    value from coverArtUrlLarge is preferred.
+
+    Maps to MPRIS property 'mpris:artUrl'.
+*/
+
+/*!
+    \qmlproperty var MprisMetaData::coverArtUrlLarge
+    \brief Large version of a cover image for the media
+
+    An url for a large cover image of the media. If defined, will
+    override coverArtUrlSmall.
+
+    Maps to MPRIS property 'mpris:artUrl'.
+*/
+
+/*!
+    \qmlproperty var MprisMetaData::albumTitle
+    \brief Title of the album of the media
+
+    The name of an album the media is a part of, or undefined
+    if not applicable or known.
+    Maps to MPRIS property 'xesam:album'.
+*/
+
+/*!
+    \qmlproperty var MprisMetaData::albumArtist
+    \brief Artist of the album of the media
+
+    The artist of an album the media is part of.
+    Maps to MPRIS property 'xesam:albumArtist'.
+*/
+
+/*!
+    \qmlproperty var MprisMetaData::contributingArtist
+    \brief Artist or artists who contributed the media
+
+    The artist or artists who have made the media.
+    Eg. band, DJ, singer, director.
+    Maps to MPRIS property 'xesam:artist'.
+*/
+
+/*!
+    \qmlproperty var MprisMetaData::lyrics
+    \brief Lyrics of the media
+
+    Maps to MPRIS property 'xesam:asText'.
+*/
+
+/*!
+    \qmlproperty var MprisMetaData::audioBpm
+    \brief Beats per minute of audio
+
+    Maps to MPRIS property 'xesam:audioBPM'.
+*/
+
+/*!
+    \qmlproperty var MprisMetaData::autoRating
+    \brief An automatic rating for the media
+
+    A rating for the media decided via automated means in the range [0, 1].
+
+    Maps to MPRIS property 'xesam:autoRating'.
+*/
+
+/*!
+    \qmlproperty var MprisMetaData::comment
+    \brief A list of free form comments about the media.
+
+    Maps to MPRIS property 'xesam:comment'.
+*/
+
+/*!
+    \qmlproperty var MprisMetaData::composer
+    \brief List of composers of the media.
+
+    Maps to MPRIS property 'xesam:composer'.
+*/
+
+/*!
+    \qmlproperty var MprisMetaData::year
+    \brief The year the media was produced.
+
+    Convenience accessor for the year part of the date property.
+    \sa MprisMetaData::date
+*/
+
+/*!
+    \qmlproperty var MprisMetaData::date
+    \brief The date the media was produced.
+
+    Maps to MPRIS property 'xesam:contentCreated'.
+*/
+
+/*!
+    \qmlproperty var MprisMetaData::chapterNumber
+    \brief
+
+    Maps to MPRIS property 'xesam:discNumber'
+*/
+
+/*!
+    \qmlproperty var MprisMetaData::firstUsed
+    \brief The date the media was played for the first time.
+
+    Maps to MPRIS property 'xesam:firstUsed'.
+*/
+
+/*!
+    \qmlproperty var MprisMetaData::genre
+    \brief The genres the media represents.
+
+    Maps to MPRIS property 'xesam:genre'.
```

Comment:
Wonder how the typing goes. Seeing a property defined like this I could assume setting a string, but the d-bus spec says list of strings.

Should we have the expected types on these in addition to the var?

Reply:
Could add the expected types, yes. But also as the typing is done internally already, I wouldn't want the developers feel the need to do unnecessary work to ensure the correct type.

Reply:
Was thinking if the "The genres" is enough to indicate that it can contain multiple ones. And even if the plural is noticed, the app developer could think if it's a comma separated list. But looking at the property set, perhaps the typing is mostly affecting this, rest should be clearer.

Reply:
Changed to list of genres, like the other possibly-multi-valued fields, better?